### ISSUE TITLE
network scale out to fit + allow zoom

### ISSUE THREAD

**Issue by cgreene** [greenelab/adage-server#129]

**Comment by mhuyck** [comment]
I believe this is related to #70, but not exactly a duplicate. The issue here is that the network itself should rescale to show the whole thing rather than get crowded around the edges as it does now. I'd like to see it auto-scale initially to show the whole network. @cgreene points out that users will need to be able to zoom in on particular areas:

1. to focus on a particular cluster of interest
2. to be able to select nodes and edges that otherwise might shrink too small to hit reliably when zoomed out

**Comment by dongbohu** [comment]
`d3.network` library doesn't have zooming option. We may need to add some options from d3 directly.

**Comment by mhuyck** [comment]
Fair enough. The features we need are setting the size of what SVG calls the "view port" and adjusting the scaling modes for the SVG object. Depending upon how d3 handles things, we may also have to manually re-scale the coordinate system. None of those things are terribly hard, but figuring out how to do them in a way that will play nice in a robust fashion will take some work.

**Comment by cgreene** [comment]
Not in d3.network library but:
http://bl.ocks.org/eyaler/10586116

**Comment by dongbohu** [comment]
Thanks!

**Comment by mhuyck** [comment]
The way that example works is very nice. The only feature I'd like to see us add would be a way to "reset" back to the original zoom level.
